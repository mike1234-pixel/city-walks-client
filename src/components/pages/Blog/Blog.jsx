import { useEffect, useContext } from "react"
import { Link } from "react-router-dom"
import PopUp from "../../PopUp/PopUp"
import { LoginContext } from "../../../context/LoginContext"
import { MDBBtn, MDBIcon, MDBCard, MDBCol, MDBCardBody, MDBCardImage, MDBCardText, MDBCardTitle} from  "mdbreact"
import urlify from "../../../functions/urlify"
import "./Blog.css"

const Blog = (props) => {

    const { popupVisible } = useContext(LoginContext)

    useEffect(() => {
        window.scrollTo(0, 0);
      });

    const { blogPosts } = props;

    const removeMarkdown = (markup) => {
        return markup.replace(/\**/g, "").replace(/#/g, "")
      }

    return (
        <div className="min-page-height">
            {popupVisible && <PopUp/>}
            <div className="page-heading-container">
                <h1 className="page-heading">Blog</h1>
            </div>
            <div className="card-container">
            {blogPosts.map((post) => {
                return (
                    <MDBCol key={post._id}>
                    <MDBCard className="blog-card">
                        <MDBCardImage className="cutter img-fluid" src={post.img} alt={post.title} waves/>
                        <MDBCardBody>
                        <MDBCardTitle className="display-font">{post.title}</MDBCardTitle>
                        <MDBCardText>{removeMarkdown(post.content.slice(0,199) + "...")}</MDBCardText>
                        <Link to={'/blog/'+urlify(post.title)}><MDBBtn outline color="elegant" className="city-card-btn">Read <MDBIcon icon="book-open" /></MDBBtn></Link>
                        </MDBCardBody>
                    </MDBCard>
                    </MDBCol>
                )
            }).reverse()}
            </div>
        </div>
    )
}

export default Blog

