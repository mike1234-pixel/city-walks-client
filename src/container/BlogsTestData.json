[{"_id":"6002e7176871ce09942e10d3","title":"Lorem Ipsum Dolor Sit Amet Consectetuer Adipiscing Elit","content":"Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc, ","img":"https://images.pexels.com/photos/1578332/pexels-photo-1578332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260","__v":0},{"_id":"600300326db5ed0d90a135ed","title":"Blogpost2","content":"Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc, ","img":"https://images.pexels.com/photos/2190283/pexels-photo-2190283.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260","__v":0},{"_id":"600300466db5ed0d90a135ee","title":"Blogpost2","content":"Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc, ","img":"https://images.pexels.com/photos/1578332/pexels-photo-1578332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260","__v":0},{"_id":"600300596db5ed0d90a135ef","title":"Blogpost4","content":"Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt. Duis leo. Sed fringilla mauris sit amet nibh. Donec sodales sagittis magna. Sed consequat, leo eget bibendum sodales, augue velit cursus nunc, ","img":"https://images.pexels.com/photos/1578332/pexels-photo-1578332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260","__v":0}]