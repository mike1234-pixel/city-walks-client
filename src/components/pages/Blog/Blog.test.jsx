import renderer from 'react-test-renderer'
import BlogsTestData from "../../../container/BlogsTestData.json"
import { BrowserRouter } from 'react-router-dom';
import Blog from './Blog'

// snapshot test
it('Blogs Page renders correctly', () => {  

  const blogPosts = BlogsTestData

  const tree = renderer
    .create(
        <BrowserRouter>
          <Blog blogPosts={blogPosts}/>
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});