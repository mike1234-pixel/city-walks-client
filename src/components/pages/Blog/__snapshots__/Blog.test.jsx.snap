// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blogs Page renders correctly 1`] = `
<div
  className="min-page-height"
>
  <div
    className="page-heading-container"
  >
    <h1
      className="page-heading"
    >
      Blog
    </h1>
  </div>
  <div
    className="card-container"
  >
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card blog-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Lorem Ipsum Dolor Sit Amet Consectetuer Adipiscing Elit"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1578332/pexels-photo-1578332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Lorem Ipsum Dolor Sit Amet Consectetuer Adipiscing Elit
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec q...
          </p>
          <a
            href="/blog/lorem-ipsum-dolor-sit-amet-consectetuer-adipiscing-elit"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Read 
              <i
                className="fa fa-book-open"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card blog-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Blogpost2"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/2190283/pexels-photo-2190283.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Blogpost2
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec q...
          </p>
          <a
            href="/blog/blogpost2"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Read 
              <i
                className="fa fa-book-open"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card blog-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Blogpost2"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1578332/pexels-photo-1578332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Blogpost2
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec q...
          </p>
          <a
            href="/blog/blogpost2"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Read 
              <i
                className="fa fa-book-open"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card blog-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Blogpost4"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1578332/pexels-photo-1578332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Blogpost4
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec q...
          </p>
          <a
            href="/blog/blogpost4"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Read 
              <i
                className="fa fa-book-open"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
